source/rust_verify/src/erase.rs

```diff
@@ -826,6 +832,96 @@ fn erase_stmt(
     kind.map(|kind| Stmt { id, kind, span })
 }
 
+fn erase_inv_block(ctxt: &Ctxt, mctxt: &mut MCtxt, expect: Mode, block: &Block) -> Block {
+    if block.stmts.len() != 3 {
+        panic!(
+            "erase_inv_block: the checks in invariant_block_to_vir should ensure the block has exactly 3 statements"
+        );
+    }
+
+    let mut result_stmts = Vec::new();
+
+    // First statement:
+    // let (guard, mut ident) = open_invariant_begin(arg);
+
+    // first, pattern match to get at the arg
+
+    let open_stmt = &block.stmts[0];
+    match &open_stmt.kind {
+        StmtKind::Local(local) => {
```

Comment:
When you have only one pattern, you can also use

```
if let StmtKind::Local(local) = &open_stmt.kind {
  ...
} else {
  panic!(...)
}
```

No need to change this, just a heads up.